apps/studio/src/components/ResourceSelector/ResourceSelector.tsx

```diff
@@ -1,41 +1,42 @@
-import { useState } from "react"
+import { Suspense, useState } from "react"
 import {
   Box,
   Flex,
   HStack,
   Icon,
-  InputGroup,
-  InputLeftElement,
+  Skeleton,
   Spacer,
   Text,
 } from "@chakra-ui/react"
-import { Button, Input, Link } from "@opengovsg/design-system-react"
-import { BiHomeAlt, BiLeftArrowAlt, BiSearch } from "react-icons/bi"
+import { Button, Link } from "@opengovsg/design-system-react"
+import { BiHomeAlt, BiLeftArrowAlt } from "react-icons/bi"
 
 import { trpc } from "~/utils/trpc"
 import { ResourceItem } from "./ResourceItem"
 
-interface ResourceSelectorProps {
-  siteId: string
-  selectedResourceId?: string
-  onChange: (resourceId: string) => void
-  isDisabledFn?: (resourceId: string) => boolean
-}
-
-export const ResourceSelector = ({
+const SuspensableResourceSelector = ({
   siteId,
   selectedResourceId,
   onChange,
   isDisabledFn,
 }: ResourceSelectorProps) => {
-  const [parentIdStack, setParentIdStack] = useState<string[]>([])
-  const [searchQuery, setSearchQuery] = useState("")
-  const currResourceId = parentIdStack[parentIdStack.length - 1]
-  const { data, fetchNextPage, hasNextPage, isFetchingNextPage } =
-    trpc.resource.getChildrenOf.useInfiniteQuery(
+  const [ancestryStack] = trpc.resource.getAncestryOf.useSuspenseQuery({
+    siteId,
+    resourceId: selectedResourceId,
+  })
+  const [parentIdStack, setParentIdStack] = useState<string[]>(
+    ancestryStack
+      .map((item) => item.id)
+      .reverse()
+      .slice(0, -1),
+  )
```

Comment:
since this is the only use case, why not return the shape directly instead of modifying it more on the client?

Reply:
Adjusted in cd38265